kernel.go

```diff
@@ -373,6 +379,82 @@ func handleExecuteRequest(ir *classic.Interp, receipt msgReceipt) error {
 	return receipt.Reply("execute_reply", content)
 }
 
+// doEval evaluates the code in the interpreter. This function captures an uncaught panic
+// as well as the values of the last statement/expression.
+func doEval(ir *classic.Interp, code string) (_ []interface{}, err error) {
+	// Capture a panic from the evaluation if one occurs and store it in the `err` return parameter.
+	defer func() {
+		if r := recover(); r != nil {
+			var ok bool
+			if err, ok = r.(error); !ok {
+				err = errors.New(fmt.Sprint(r))
+			}
+		}
+	}()
+
+	// Prepare and perform the multiline evaluation.
```

Comment:
It's more of a preference thing, but I like to keep a white space before each comment (i.e., not nestled up against other code). Nitpick